src/main/java/br/com/caelum/comidinhas/handler/validation/ValidationErrorHandler.java

```diff
@@ -0,0 +1,49 @@
+package br.com.caelum.comidinhas.handler.validation;
+
+import br.com.caelum.comidinhas.exception.RelationshipNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.*;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestControllerAdvice
+public class ValidationErrorHandler {
+    @Autowired
+    private MessageSource messageSource;
```

Comment:
Atributo de dependência private e final recebido no construtor, ao invés de `@Autowired`.
Onde puder injetar pelo construtor, vamos injetar.

```suggestion
    private final  MessageSource messageSource;
    
     ValidationErrorHandler(MessageSource messageSource) {
        this.messageSource = messageSource;
    }
```